pytest-embedded-idf/pytest_embedded_idf/serial.py

```diff
@@ -207,15 +198,12 @@ def dump_flash(
         else:
             raise ValueError('You must specify "partition" or ("address" and "size") to dump flash')
 
-        content = self.stub.read_flash(_addr, _size)
         if output:
-            if isinstance(output, str):
-                os.makedirs(os.path.dirname(output), exist_ok=True)
-                with open(output, 'wb') as f:
-                    f.write(content)
-            else:
-                output.write(content)
+            esptool.main(['read_flash', str(_addr), str(_size), str(output)], esp=self.esp)
         else:
+            with tempfile.NamedTemporaryFile() as fp:
+                esptool.main(['read_flash', str(_addr), str(_size), fp.name], esp=self.esp)
+                content = fp.read()
```

Comment:
or use stringIO as a io buffer

Reply:
I used NamedTemporaryFile because esptool requires a file name.

Reply:
good to know. Thank you!